arangod/Transaction/Methods.cpp

```diff
@@ -2644,18 +2644,17 @@ Future<OperationResult> transaction::Methods::truncateLocal(
       VPackObjectBuilder ob(&body);
       body.add("collection", collectionName);
     }
-    try {
-      leaderState->replicateOperation(
-          body.sharedSlice(),
-          replication2::replicated_state::document::OperationType::kTruncate,
-          state()->id(), trxColl->collectionName(),
-          replication2::replicated_state::document::ReplicationOptions{});
-    } catch (basics::Exception const& e) {
-      return OperationResult(Result{e.code(), e.what()}, options);
-    } catch (std::exception const& e) {
-      return OperationResult(Result{TRI_ERROR_INTERNAL, e.what()}, options);
-    }
-    return OperationResult{Result{}, options};
+    auto operation =
+        replication2::replicated_state::document::ReplicatedOperation::
+            buildTruncateOperation(state()->id(), trxColl->collectionName());
+    // Should finish immediately
+    auto replicationRes =
+        leaderState
+            ->replicateOperation(
+                std::move(operation),
+                replication2::replicated_state::document::ReplicationOptions{})
+            .get();
+    return OperationResult{replicationRes.result(), options};
```

Comment:
Why should this finish immediately? Shouldn't this be async?

Reply:
It finishes immediately because we are not waiting for the operation to be committed in the log. I added an assertion and comment on that.